### PR TITLE
adds v2 alert endpoints

### PR DESCRIPTION
**Pull request by shalper**
# What changed

Added v2/alert endpoints: 
https://deploy-preview-745--logz-docs.netlify.app/api/#tag/Alerts

### CODE REVIEW

File: _source/api/logzio-public-api.yml [hunk 1]
```diff
@@ -4813,3 +4932,369 @@ definitions:
...
+          - SUM
+          - MIN
+          - MAX
+          - AVG
+          - COUNT
+          - UNIQUE_COUNT
+          - NONE
+      fieldToAggregateOn:
+        type: string
+        description: Selects the field on which to run the aggregation for the trigger condition.
```
Comment (AmiStrn) [inline _source/api/logzio-public-api.yml:4955]: We have a bunch of conditions regarding the aggregations and group by (nothing new)
1. fieldToAggregateOn can't be in the group by fields.
2. aggregationType is COUNT -> fieldToAggregateOn must be null, and group by fields must not be empty
3. aggregationType is NONE -> fieldToAggregateOn must be null, and group by fields must be empty (or null)
4. aggregationType is any of the other types -> fieldToAggregateOn must not be null

File: _source/api/logzio-public-api.yml [hunk 2]
```diff
@@ -4813,3 +4932,369 @@ definitions:
...
+        items:
+          $ref: '#/definitions/SubAlert'
+      correlations:
+        $ref: '#/definitions/SubAlertCorrelation'
+      enabled:
+        type: boolean
+        description: If `true`, the alert is enabled and active.
+  AlertTrigger:
+    type: object
+    description: Sets an additional triggering threshold and severity tab to label the event when the alert triggers.
```
Comment (AmiStrn) [inline _source/api/logzio-public-api.yml:5201]: it is **the** condition for which the alert triggers, so the word "additional" makes it sound not required somehow.
```suggestion
    description: Sets the triggering threshold and severity tab to label the event when the alert triggers.
```

File: _source/api/logzio-public-api.yml [hunk 3]
```diff
@@ -4813,3 +4932,369 @@ definitions:
...
+      severityThresholdTiers:
+        type: object
+        description: >-
+          Sets a severity label per trigger threshold as a key:value pair.
+          
+          
+          If using more than one sub-component, only 1 severityThresholdTiers is allowed. Otherwise, 1 per `enum` are allowed (for a total of 5 thresholds of increasing severities).
+
+
+          Increasing severity must adhere to the logic of the operator.
```
Comment (AmiStrn) [inline _source/api/logzio-public-api.yml:5223]: nice! that is very clear

File: _source/api/logzio-public-api.yml [hunk 4]
```diff
@@ -4813,3 +4932,369 @@ definitions:
...
+    type: object
+    description: Customize the alert output to be sent out in notifications when the alert triggers.
+    properties:
+      fieldName:
+        type: string
+        description: Specify the fields to be included in the notification.
+      regex:
+        type: string
+        description: Trims the data using regex filters. [Learn more](https://docs.logz.io/user-guide/alerts/regex-filters.html)
+      sort:
```
Comment (AmiStrn) [inline _source/api/logzio-public-api.yml:5267]: an exception is thrown if the user selects an "analyzed field" to be the fieldName AND sort to be either of the enum values. in other words - you cant sort by an analyzed field.

File: _source/api/logzio-public-api.yml [hunk 5]
```diff
@@ -4813,3 +4932,382 @@ definitions:
...
+          type: string
+          example: tom.a@logz.io
+      notificationEndpointIds:
+        type: array
+        description: Array of IDs of pre-configured endpoint channels to notify when the alert triggers.
+        items:
+          type: integer
+          format: int32
+          example: 10101
+  SubAlert:
```
Comment (AmiStrn) [inline _source/api/logzio-public-api.yml:5113]: i see that the name SubAlert appears in the doc, although not required in any of the fields in order to create the api request. 
Is this an issue? should we change the naming asap?

File: _source/api/logzio-public-api.yml (hunk 2 above)
Comment (shalper) [reply _source/api/logzio-public-api.yml:5214]: fixed

File: _source/api/logzio-public-api.yml (hunk 4 above)
Comment (shalper) [reply _source/api/logzio-public-api.yml:5515]: Updated the description.

File: _source/user-guide/alerts/index.md [hunk 6]
```diff
@@ -15,7 +15,7 @@ Log alerts can ensure that you're notified of critical events. Configuring the r
 
 Logz.io alerts use a Kibana search query to continuously scan your logs and alert you when a certain set of conditions is met. The simplest alerts can use a simple search query or a particular filter, but others can be quite complex and involve several conditions with varying thresholds.
 
-Event logs of triggered alerts are always available and searchable in Kibana. In Kibana Discover, just filter for `logzio-alert:exists`. But you also have the option to add notifications, and control their contents, format, and who they are sent to.
+Event logs of triggered alerts are always available and searchable in Kibana. In Kibana Discover, just filter for `_exists_:logzio-alerts`. But you also have the option to add notifications, and control their contents, format, and who they are sent to.
```
Comment (shalper) [inline _source/user-guide/alerts/index.md:18]:
```suggestion
Event logs of triggered alerts are always available and searchable in Kibana. In Kibana Discover, just filter for `_exists_:logzio-alert`. But you also have the option to add notifications, and control their contents, format, and who they are sent to.
```

**Pull request merged**